hadoop-hdfs-project/hadoop-hdfs/src/main/java/io/hops/transaction/lock/LeaseLock.java

```diff
@@ -45,9 +47,17 @@
 
   @Override
   protected void acquire(TransactionLocks locks) throws IOException {
+
+    setLockMode(lockType);
+
     Set<String> hldrs = new HashSet<>();
-    if (leaseHolder != null) {
+    if (leaseHolder != null && singleFileLock == null) { //normal operation. lock lease and all paths
       hldrs.add(leaseHolder);
+    } else if (leaseHolder != null && singleFileLock != null) {
+      Lease lease = acquireLock(lockType, Lease.Finder.ByHolder, leaseHolder, Lease.getHolderId(leaseHolder));
```

Comment:
Can you explain what problem this is solving? The jira does not tel much about leases.

Reply:
The clinet has lots of open files. When the client opens a new file then previous we took a lock on all the open leases for that client. This was slow as the client may have millions of files open. That is one of the reason why the bechmark was slow